Controllers/ProductController.cs

```diff
@@ -0,0 +1,130 @@
+﻿using EFCoreServices.Models;
+using EFCoreServices.Repository;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace EFCoreServices.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ProductController : ControllerBase
+    {
+        IProductRepository productRepository;
+        public ProductController(ProductRepository _productRepository)
+        {
+            productRepository = _productRepository;
+        }
+
+        [HttpGet]
+        [Route("GetProductsList")]
+        public async Task<IActionResult> GetProductsList()
+        {
+            try
+            {
+                var products = await productRepository.GetProductsList();
+                if(products == null)
+                {
+                    return NotFound();
+                }
+                return Ok(products);
+            }
+            catch(Exception)
+            {
+                return BadRequest();
```

Comment:
All exceptions should be logged. 
BadRequest means the user input was bad. which I don't think is true in this case. 
you can use 5xx error codes like internal server error or service unavailable and return the exception message